main.go

```diff
@@ -1,33 +1,35 @@
+/***************************************************************************
+ *
+ * Copyright (c) 2020 Bonc.com.cn, Inc. All Rights Reserved
+ *
+ **************************************************************************/
+
+/**
+ * @file    main_test.go.go
+ * @author  qihouliang(qihouliang@bonc.com.cn)
+ * @date    2023/4/11 22:26
+ * @brief
+ */
+
 package main
 
 import (
-	"context"
-	"os"
-	"os/signal"
-	"syscall"
-
+	"fmt"
 	"github.com/appleboy/CodeGPT/cmd"
+	"github.com/appleboy/CodeGPT/webhook"
+	"github.com/spf13/viper"
+	"log"
+	"net/http"
 )
 
-func withContextFunc(ctx context.Context, f func()) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
-	go func() {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
-		defer signal.Stop(c)
-
-		select {
-		case <-ctx.Done():
-		case <-c:
-			cancel()
-			f()
-		}
-	}()
-
-	return ctx
-}
-
 func main() {
-	ctx := withContextFunc(context.Background(), func() {})
-	cmd.Execute(ctx)
+	log.SetFlags(log.Lshortfile | log.LstdFlags)
+	cmd.InitConfig()
+	http.HandleFunc("/payload", webhook.HandleWebhook)
+	ip := viper.GetString("server.ip")
+	port := viper.GetString("server.port")
+	var address = fmt.Sprintf("%s:%s", ip, port)
+	fmt.Println("Listening for GitHub Webhooks on", address)
+	http.ListenAndServe(address, nil)
+
 }
```

Comment:
Overall, the code patch looks relatively simple and straightforward. Here are a few potential issues and suggestions for improvement:

1. Security: The code seems to be listening for incoming requests on the "/payload" endpoint without any authentication or authorization mechanism. This could potentially lead to security vulnerabilities, such as unauthorized access or injection attacks. It would be a good idea to implement some form of authentication and authorization before allowing any incoming requests.

2. Error handling: The code does not appear to have any error handling mechanisms, which could lead to unexpected crashes or behavior. It would be a good idea to implement some error handling, such as logging or returning appropriate error messages to clients.

3. Code structure: The code appears to be organized in a single main function, which could become difficult to manage as the codebase grows. It might be a good idea to split the code into smaller, more manageable functions/modules.

4. Testing: The code does not appear to have any unit tests or integration tests. Adding tests can help catch bugs and ensure that the code behaves as expected.

5. Code documentation: The code lacks comments and documentation, which can make it difficult for other developers to understand the code. It would be helpful to add comments and documentation to clarify the purpose and functionality of the code.